pilot/pkg/networking/util/util.go

```diff
@@ -137,24 +128,48 @@ func ListContains(haystack []string, needle string) bool {
 
 // ConvertAddressToCidr converts from string to CIDR proto
 func ConvertAddressToCidr(addr string) *core.CidrRange {
-	if len(addr) == 0 {
-		return nil
+	cidr, err := AddrStrToCidrRange(addr)
+	if err != nil {
+		log.Errorf("failed to convert address %s to CidrRange: %v", addr, err)
 	}
 
-	cidr := &core.CidrRange{
-		AddressPrefix: addr,
-		PrefixLen: &wrapperspb.UInt32Value{
-			Value: getMaxCidrPrefix(addr),
-		},
+	return cidr
+}
+
+// AddrStrToCidrRange converts from string to CIDR proto
+func AddrStrToCidrRange(addr string) (*core.CidrRange, error) {
```

Comment:
since there were unit tests for  this method previously, can you add few test cases around this ?

Reply:
revered the previous test cases